source-map.bs

```diff
@@ -674,22 +676,84 @@ The index map follows the form of the standard map.  Like the regular source map
 the file format is JSON with a top-level object.  It shares the [=json/version=] and
 [=json/file=] field from the regular source map, but gains a new [=sections=] field.
 
-<dfn><code>sections</code></dfn> is an array of [=Section=] objects.
+<dfn dfn-for="index-map"><code>sections</code></dfn> is an array of [=Section=] objects.
 
 ## Section
 
 Section objects have the following fields:
 
-* <dfn><code>offset</code></dfn> is an object with two fields, `line` and `column`,
+* <dfn dfn-for="index-map"><code>offset</code></dfn> is an object with two fields, `line` and `column`,
     that represent the offset into generated code that the referenced source map
     represents.
 
-* <dfn><code>map</code></dfn> is an embedded complete source map object.
+* <dfn dfn-for="index-map"><code>map</code></dfn> is an embedded complete source map object.
     An embedded map does not inherit any values from the containing index map.
 
 The sections must be sorted by starting position and the represented sections
 must not overlap.
 
+## Decoding an Index Map
+
+To <dfn export>decode an index source map</dfn> given a [=string=]-keyed [=/map=] |jsonMap| and a [=/URL=]
+|baseURL|, run the following steps:
+1. <p>Assert: |jsonMap|[`"sections"`] [=map/exists=].</p>
+1. If |jsonMap|[`"sections"`] is not a [=list=], report an error.
+1. If |jsonMap|[`"version"`] does not [=map/exist=] or |jsonMap|[`"version"`] is not 3,
+    [=optionally report an error=].
+1. Let |sourceMap| be a new [=decoded source map=].
+1. Set |sourceMap|'s [=decoded source map/file=] to [=optionally get a string=] `"file"` from |jsonMap|.
+1. Let |previousOffset| be null.
+1. Let |previousLastMapping| be null.
+1. [=For each=] |section| of |jsonMap|[`"sections"`]:
+    1. If |section| is not a [=/map=], [=optionally report an error=] and continue with the next iteration.
+    1. If |section|[`"offset"`] does not [=map/exist=], then report an error.
+    1. Let |offsetLine| be |section|[`"offset"`][`"line"`].
+    1. Let |offsetColumn| be |section|[`"offset"`][`"column"`].
+    1. If |offsetLine| is not a number, [=optionally report an error=].
+    1. If |offsetColumn| is not a number, [=optionally report an error=].
+    1. If |previousOffset| is not null,
+        1. If |previousLastMapping| is not null,
+            1. If |offsetLine| is less than |previousLastMapping|'s [=decoded mapping/generatedLine=], [=optionally report an error=].
```

Comment:
Should these range checks could be "less than or equal to" to be strict inequality rather than allowing equality?